### DISCUSSION TITLE
Media player entity without next and previous button

### DISCUSSION THREAD

**Discussion by ady1972** [Blackymas/NSPanel_HA_Blueprint#1221] [Q&A]
Hi all,
i'm usign this fantastic mod for NSPanel and i'm very satisfied !!!
I have a question.
I'm using an automation that will wake up display when my Google group of soiundspeaker start playing music showing page of media player entity.
Everything works well concerning automation, but htere is a little problem.
When i use Spotify to play musci everything is perfect , but when i use internale music (form home assistant or from plex media server throught Music Assistant add-on) button for next and previous song don't appear.
Is it normal?  Is there any solution for it?

If not, because i have modified some fonts and icon size of the panel using Nextion editor, i have tought to insert two static button that will send command media_player.media_prevoius_track (or next track) to be able to change  song directly from display but i have no idea on how to do it. Is there any link for it to read a guide?

Thanks in advance

Ciao

![IMG_20231102_181307328](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/728bed03-8be6-4bc6-bef8-b26ddf997750)
![IMG_20231102_181339360](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/dfb19442-3eea-4bc8-8fd1-11e3e1b6a91f)

**Comment by edwardtfn** [comment]
The buttons are shown based on the features reported by the entity in Home Assistant.
Could you please open Developer Tools, go to tab "States" and find the entity related to this, then share here it's attributes? I'm specially interested in `supported_features` attribute.
Please make sure you hide any sensitive info before sharing.

**Comment by ady1972** [comment]
First of all thanks for quick reply.
I hope to have understood well your request :).
In attached picture of status of media player entity.
However i notice that entity card in home assistant has next and previuos button.

![Screenshot 2023-11-03 143736](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/7cc3102c-ae0c-4a95-8d42-233573dfc545)
![Screenshot 2023-11-03 143809](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/10bc9777-280a-4204-a18f-54629f177b3d)

**Reply by edwardtfn** [reply]
`supported_features: 152463`, which is not including previous/next track:

```
 => PAUSE = 1
 => SEEK = 2
 => VOLUME_SET = 4
 => VOLUME_MUTE = 8
    PREVIOUS_TRACK = 16
    NEXT_TRACK = 32

 => TURN_ON = 128
 => TURN_OFF = 256
 => PLAY_MEDIA = 512
    VOLUME_STEP = 1024
    SELECT_SOURCE = 2048
 => STOP = 4096
    CLEAR_PLAYLIST = 8192
 => PLAY = 16384
    SHUFFLE_SET = 32768
    SELECT_SOUND_MODE = 65536
 => BROWSE_MEDIA = 131072
    REPEAT_SET = 262144
    GROUPING = 524288
    MEDIA_ANNOUNCE = 1048576
    MEDIA_ENQUEUE = 2097152
```

It's weird that you have the controls in Home Assistant. Are those operational?
Which integration are you using for this device? I will try to look at their docs/code.

By the way, is this Spanish or Portuguese? 😃

**Reply by edwardtfn** [reply]
It is interesting to see that "Mini Media player" also use `supported_features` and bytes `16` and `32` to check if Next/Prev are available: https://github.com/kalkih/mini-media-player/blob/b45330ddfd9f1d6bbc31403f11acaec62021c921/src/model.ts#L178

**Comment by ady1972** [comment]
Control are operational. I use Music Assistant and card is Mini Media Player.

And...is Italian 😄

**Reply by edwardtfn** [reply]
Ciao,

May I ask you to run some tests?
With a music playing, go to Developer Tools and the tab "Services", the select service "Media Player: Next" and, as a target, select your device. Then please press "Call service" and let me know if that works for changing to the next track.
If that works, please press the button "Go to yaml mode", copy that yaml and share it here, please.

**Comment by ady1972** [comment]
Ciao

i confirm (i have already tried before psoting) that service media_player.next works perfectly. Here below yaml

service: media_player.media_next_track
data: {}
target:
  entity_id: media_player.gruppo_casa

For now i have solved problem in this way.
Working with Nextion editor i have substituted button  next and previous present in hmi file, with 2 new buttons and flag "send id command" option. Then i have added them to ESP file with this code

binary_sensor:
  - platform: nextion
    name: $device_name Next Button
    page_id: 25
    component_id: 25

Where page id is the number of page where i have created button, and component id is the number of component shown in properties of button in Nextion editor-
After updating ESP32 for NSPanel two new button appears and they can be used for automate service media_player.next.

Here is the result... I need to make some more setup about button aspect ;P

![IMG_20231104_141320145](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/f798c5e5-4002-443f-8bbd-5155d42c8636)

**Reply by edwardtfn** [reply]
> I need to make some more setup about button aspect

Set the `sta` to `crop image` and then `picc` to 0.
You can change font to 9 or 10 if you wanna that bigger and you can use the icon from [HASP icons](https://htmlpreview.github.io/?https://github.com/jobr99/Generate-HASP-Fonts/blob/master/cheatsheet.html). Click on the "U" relative to the icon you want and then paste it on the `txt` field of the component in Nextion Editor.
We don't use that way as it will break the Nextion2Text used to track the changes, but you can use like that.
By the way, you can also add this to your Home Assistant configuration:
```yaml
homeassistant:
  customize:
    media_player.gruppo_casa:
      supported_features: 152511
```
This will inform your device supports also Next/Previous track...
I still working in a fix, so hopefully this won't be needed soon.

**Comment by ady1972** [comment]
Thanks a lot for your suggest.

I've followed your mini tutorial to improve setup of new button but it remains as black rectangle, no icons is shown.
When i click on U on HASP icons you have sent this symbol "" is copied for all icon. I ve tried also to copy value ex "EE97" in the txt filed but nothing happens.
Am i doing something wrong?

**Comment by edwardtfn** [comment]
By the way, which version of Music Assistant are you running?
Can be found in: HACS -> Integrations -> Music Assistant -> Information For example: 2022.6.2

**Reply by edwardtfn** [reply]
As I understood, when creating a group, Music Assistant will use the `supported_features` from all the devices in that group, so if a single device cannot support one feature, the entire group will not have that feature flagged.
Just to help me with this troubleshooting, could you please share the `supported_features` of each player on this `media_player.grupo_casa_3`?

**Reply by ady1972** [reply]
Version of Music Assistant is 
2023.6.0b6

**Reply by ady1972** [reply]
About supported feature of devices of gruppo casa, i part of them has supported feature 3602367 and the rest  152461

**Reply by edwardtfn** [reply]
Ok, one more question, sorry...
When testing here, all my Music Assist related entities have attributes `mass_player_type`, `mass_player_id`, `group_members` and `group_leader`. I don't see those attributes in your [previous screenshot](https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1221#discussioncomment-7467328). Could you please double check if you see those in your entities?

**Reply by edwardtfn** [reply]
By the way:

```
supported_features: 3602367
=> PAUSE = 1
=> SEEK = 2
=> VOLUME_SET = 4
=> VOLUME_MUTE = 8
=> PREVIOUS_TRACK = 16
=> NEXT_TRACK = 32

=> TURN_ON = 128
=> TURN_OFF = 256
=> PLAY_MEDIA = 512
=> VOLUME_STEP = 1024
   SELECT_SOURCE = 2048
=> STOP = 4096
=> CLEAR_PLAYLIST = 8192
=> PLAY = 16384
=> SHUFFLE_SET = 32768
   SELECT_SOUND_MODE = 65536
=> BROWSE_MEDIA = 131072
=> REPEAT_SET = 262144
   GROUPING = 524288
=> MEDIA_ANNOUNCE = 1048576
=> MEDIA_ENQUEUE = 2097152
```

Which is almost everything. 😉 

And:
```
supported_features: 152461
=> PAUSE = 1
   SEEK = 2
=> VOLUME_SET = 4
=> VOLUME_MUTE = 8
   PREVIOUS_TRACK = 16
   NEXT_TRACK = 32

=> TURN_ON = 128
=> TURN_OFF = 256
=> PLAY_MEDIA = 512
   VOLUME_STEP = 1024
   SELECT_SOURCE = 2048
=> STOP = 4096
   CLEAR_PLAYLIST = 8192
=> PLAY = 16384
   SHUFFLE_SET = 32768
   SELECT_SOUND_MODE = 65536
=> BROWSE_MEDIA = 131072
   REPEAT_SET = 262144
   GROUPING = 524288
   MEDIA_ANNOUNCE = 1048576
   MEDIA_ENQUEUE = 2097152
```
Which is the one limiting the `Next`/`Prev` buttons.

**Comment by ady1972** [comment]
In attached yaml code of my gruppo casa.

![Screenshot 2023-11-06 170803](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/7114188/431d46a7-9e4a-4a2c-92fc-02c4d0d698dd)

**Reply by edwardtfn** [reply]
Ok, thanks!
I believe I have everything I need to work in a fix. Just have to find some time now. 😂
I come back to you.

**Reply by ady1972** [reply]
Take your time.... You've already done a great and fantastic work 💪💪💪

Thanks a lot for this

**Reply by edwardtfn** [reply]
@ady1972, I believe this is solved on `dev` and hopefully will be included in the next release.
Unfortunately I cannot duplicate the problem, so I cannot validate the solution.
If you wanna give a try in `dev` please let me know.

**Reply by ady1972** [reply]
GReat job...It could be interesting to try...where can i find dev version?
I just have a second display that i use for experiment that could be useful for this !!! ;P

**Reply by ady1972** [reply]
Found it....i'll give it a try
